apisprout.go

```diff
@@ -510,9 +514,22 @@ func server(cmd *cobra.Command, args []string) {
 					AuthenticationFunc: func(c context.Context, input *openapi3filter.AuthenticationInput) error {
 						// TODO: support more schemes
 						sec := input.SecurityScheme
-						if sec.Type == "http" && sec.Scheme == "bearer" {
-							if req.Header.Get("Authorization") == "" {
-								return ErrMissingAuth
+						if sec.Type == "http" {
+							// Prefixes for each scheme.
+							prefixes := map[string]string{
+								"bearer": "Bearer ",
+								"basic":  "Basic ",
+							}
+							if prefix, ok := prefixes[sec.Scheme]; ok {
+								auth := req.Header.Get("Authorization")
+								// If the auth is missing
+								if len(auth) == 0 {
+									return ErrMissingAuth
+								}
+								// If the auth doesn't have a value or doesn't start with the prefix
+								if len(auth) <= len(prefix) || !strings.HasPrefix(auth, prefix) {
```

Comment:
It looks like the token type is case insensitive, so we need to handle that here:

> The type of the token issued as described in Section 7.1.  Value is case insensitive.
https://tools.ietf.org/html/rfc6749#section-5.1

Also see:

> It uses an extensible case-insensitive token to identify the authentication scheme
https://tools.ietf.org/html/rfc2617#section-1.2

Reply:
This should be fixed. I made the prefixes upper case and called ToUpper on the HTTP header's value.